airflow/models/dataset.py

```diff
@@ -0,0 +1,75 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from urllib.parse import urlparse
+
+from sqlalchemy import Column, Index, Integer, String
+
+from airflow.models.base import Base
+from airflow.utils import timezone
+from airflow.utils.sqlalchemy import ExtendedJSON, UtcDateTime
+
+
+class Dataset(Base):
+    """
+    A table to store datasets.
+
+    :param uri: a string that uniquely identifies the dataset
+    :param extra: JSON field for arbitrary extra info
+    """
+
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    uri = Column(
+        String(length=3000).with_variant(
+            String(
+                length=3000,
+                # latin1 allows for more indexed length in mysql
+                # and this field should only be ascii chars
+                collation='latin1_general_cs',
+            ),
+            'mysql',
+        ),
+        nullable=False,
+    )
```

Comment:
```suggestion
        nullable=False,
        unique=True,
    )
```

since you are passing the same via Index on L54

Reply:
so i _thought_ that we manage uniqueness just through the manually-created index, so  that we can give it a  name (so that we can manually deal with it in migrations if necessary) -- i don't know what happens if we also add unique keyword here -- will it create the constraint without our provided name?  @ephraimbuddy you know what we should do  here?

Reply:
I think having `unique=True` on the column would create an implicit unnamed constraint. And, I feel the named `index` with `unique=True` would create an index as well as an unnamed unique constraint in some databases like MSSQL but I'm not sure. I will verify this

Reply:
No. It works fine in MSSQL and postgres. It didn't create extra unique constraint. Just a unique index.

Reply:
so does it result in the creation of two indexes? or  just one?   cus we don't want  to create two of them.